precise_driver/CMakeLists.txt

```diff
@@ -4,22 +4,40 @@ project(precise_driver)
 add_compile_options(-std=c++11)
 
 find_package(catkin REQUIRED COMPONENTS
+  actionlib_msgs
   angles
   cob_srvs
+  controller_manager_msgs
   geometry_msgs
+  message_generation
   realtime_tools
   ros_control_boilerplate
+  std_msgs
   std_srvs
 )
 
+add_service_files(
+  FILES
+  Gripper.srv
+  Plate.srv
+)
+
+generate_messages(
+  DEPENDENCIES
+  actionlib_msgs
+  controller_manager_msgs
+  geometry_msgs
+  std_msgs
+)
+
 catkin_package(
-  CATKIN_DEPENDS cob_srvs geometry_msgs std_srvs
+  CATKIN_DEPENDS actionlib_msgs cob_srvs controller_manager_msgs geometry_msgs message_runtime std_msgs std_srvs
 )
 
 ## Build
 include_directories(include ${catkin_INCLUDE_DIRS})
 
-add_library(precise_hw_interface src/pflex_device.cpp src/precise_hw_interface.cpp src/precise_tcp_interface.cpp)
+add_library(precise_hw_interface src/device/device.cpp src/device/tcp_client.cpp src/precise_hw_interface.cpp)
 add_dependencies(precise_hw_interface ${catkin_EXPORTED_TARGETS})
```

Comment:
this library has a dependency to the generated messages and thus needs `${${PROJECT_NAME}_EXPORTED_TARGETS}`, too

Reply:
also add this dependency to the executable below